src/main/java/games/strategy/engine/config/client/remote/UrlRedirectResolver.java

```diff
@@ -0,0 +1,43 @@
+package games.strategy.engine.config.client.remote;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * Module that follows URLs and any redirects and determines the final URL to use.
+ */
+class UrlRedirectResolver {
```

Comment:
> If we want to implement everything correctly, so everything is properly handled, we could directly use HttpComponents as this saves us a lot of time!

Agree. Not in scope here though. This class got involved because it needed testing along with everything else, so stuff was moved, and intentionally not updated functionally beyond what was necessary for test to avoid this branch from having yet more changes on it.